microservices/front-end/src/components/collaboration/Editor.tsx

```diff
@@ -261,6 +271,33 @@ function Editor({ socket }: EditorProps) {
           >
             Format Code
           </Button>
+
+          <Button
+            variant="contained"
+            style={{ minWidth: "150px" }}
+            onClick={() => {
```

Comment:
I think we can move the `onClick` function to a useCallback hook to reduce nesting (this also applies to the format code button at line 251)